modules/platforms/dotnet/Apache.Ignite.Core.Tests/Client/Cache/CacheClientAbstractTxTest.cs

```diff
@@ -0,0 +1,822 @@
+                    .And.Message.Contains("Cache transaction timed out"));
+                Assert.Throws(constraint, () => cache.Put(1, 10));
+                Assert.Throws(constraint, () => tx.Commit());
+            }
+
+            Assert.AreEqual(1, cache.Get(1));
+        }
+
+        /// <summary>
+        /// Tests that commit applies cache changes.
+        /// </summary>
+        [Test]
+        public void TestTxCommit()
+        {
+            var cache = GetTransactionalCache();
+
+            cache.Put(1, 1);
+            cache.Put(2, 2);
+
+            using (var tx = Client.GetTransactions().TxStart())
+            {
+                cache.Put(1, 10);
+                cache.Put(2, 20);
+
+                tx.Commit();
+            }
+
+            Assert.AreEqual(10, cache.Get(1));
+            Assert.AreEqual(20, cache.Get(2));
+        }
+
+        /// <summary>
+        /// Tests that rollback reverts cache changes.
+        /// </summary>
+        [Test]
+        public void TestTxRollback()
+        {
+            var cache = GetTransactionalCache();
+            cache.Put(1, 1);
+            cache.Put(2, 2);
+
+            using (var tx = Client.GetTransactions().TxStart())
+            {
+                cache.Put(1, 10);
+                cache.Put(2, 20);
+
+                Assert.AreEqual(10, cache.Get(1));
+                Assert.AreEqual(20, cache.Get(2));
+                tx.Rollback();
+            }
+
+            Assert.AreEqual(1, cache.Get(1));
+            Assert.AreEqual(2, cache.Get(2));
+        }
+
+        /// <summary>
+        /// Tests that closing transaction without commit reverts cache changes.
+        /// </summary>
+        [Test]
+        public void TestTxClose()
+        {
+            var cache = GetTransactionalCache();
+
+            cache.Put(1, 1);
+            cache.Put(2, 2);
+
+            using (Client.GetTransactions().TxStart())
+            {
+                cache.Put(1, 10);
+                cache.Put(2, 20);
+            }
+
+            Assert.AreEqual(1, cache.Get(1));
+            Assert.AreEqual(2, cache.Get(2));
+        }
+
+        /// <summary>
+        /// Tests that client can't start multiple transactions in one thread.
+        /// </summary>
+        [Test]
+        public void TestThrowsIfMultipleStarted()
+        {
+            TestThrowsIfMultipleStarted(
+                () => Client.GetTransactions().TxStart(),
+                () => Client.GetTransactions().TxStart());
+        }
+
+        /// <summary>
+        /// Tests that different clients can start transactions in one thread.
+        /// </summary>
+        [Test]
+        public void TestDifferentClientsCanStartTransactions()
+        {
+            Assert.DoesNotThrow(() =>
+            {
+                using (Client.GetTransactions().TxStart())
+                using (GetClient().GetTransactions().TxStart())
+                {
+                    // No-op.
```

Comment:
We should check that operations can be performed and that one client does not use the TX started by the other.
I think this is not the case, since we only have one thread local which is shared by all clients.

Reply:
Added test.